src/main/java/de/zalando/aruha/nakadi/controller/EventStreamController.java

```diff
@@ -0,0 +1,170 @@
+                                new TypeReference<ArrayList<Cursor>>() {
+                                });
+                    } catch (IOException e) {
+                        writeProblemResponse(response, outputStream, BAD_REQUEST,
+                                "incorrect syntax of X-nakadi-cursors header");
+                        return;
+                    }
+                }
+
+                // check that offsets are not out of bounds and partitions exist
+                if (cursors != null && !topicRepository.areCursorsValid(topic, cursors)) {
+                    writeProblemResponse(response, outputStream, PRECONDITION_FAILED, "cursors are not valid");
+                    return;
+                }
+
+                // if no cursors provided - read from the newest available events
+                if (cursors == null) {
+                    cursors = topicRepository
+                            .listPartitions(topic)
+                            .stream()
+                            .map(pInfo -> new Cursor(pInfo.getPartitionId(), pInfo.getNewestAvailableOffset()))
+                            .collect(Collectors.toList());
+                }
+
+                final Map<String, String> streamCursors = cursors
+                        .stream()
+                        .collect(Collectors.toMap(
+                                Cursor::getPartition,
+                                Cursor::getOffset));
+
+                final EventStreamConfig streamConfig = new EventStreamConfig(topic, streamCursors, batchLimit,
+                        streamLimit, batchTimeout, streamTimeout, streamKeepAliveLimit);
+
+                response.setStatus(HttpStatus.OK.value());
+
+                final String acceptEncoding = request.getHeader("Accept-Encoding");
+                final boolean gzipEnabled = acceptEncoding != null && acceptEncoding.contains("gzip");
+                final OutputStream output = gzipEnabled ? new FlushableGZIPOutputStream(outputStream) : outputStream;
+
+                if (gzipEnabled) {
+                    response.addHeader("Content-Encoding", "gzip");
+                }
+
+                final EventConsumer eventConsumer = topicRepository.createEventConsumer(topic,
+                        streamConfig.getCursors());
+                final EventStream eventStream = new EventStream(eventConsumer, output, streamConfig);
+                eventStream.streamEvents();
+
+                if (gzipEnabled) {
+                    output.close();
+                }
+
+            }
+            catch (final NakadiException e) {
+                writeProblemResponse(response, outputStream, SERVICE_UNAVAILABLE, e.getProblemMessage());
+            }
+            catch (final Exception e) {
+                writeProblemResponse(response, outputStream, INTERNAL_SERVER_ERROR, e.getMessage());
+            }
+            finally {
+                outputStream.flush();
```

Comment:
How do you know that?
The documentation of OutputStream.close() does not say that. And the implementation can do whatever it wants.
So I would prefer to stay on the safe side.